src/structures/NewsChannel.js

```diff
@@ -13,6 +14,25 @@ class NewsChannel extends TextChannel {
     // News channels don't have a rate limit per user, remove it
     this.rateLimitPerUser = undefined;
   }
+
+  /**
+   * Adds the target to this channel's followers.
+   * @param {GuildChannelResolvable} channel The channel where the webhook should be created
+   * @param {string} [reason] Reason for creating the webhook
+   * @returns {Promise<NewsChannel>}
+   * @example
+   * if (channel.type === 'news') {
+   *   channel.addFollower('222197033908436994', 'Important announcements')
+   *     .then(() => console.log('Added follower'))
+   *     .catch(console.error);
+   * }
+   */
+  async addFollower(channel, reason) {
+    channel = this.client.channels.resolve(channel);
+    if (!channel || !['text', 'news'].includes(channel.type)) throw new Error('GUILD_TEXT_CHANNEL_RESOLVE');
```

Comment:
```suggestion
    if (!channel || channel.type !== 'text') throw new Error('GUILD_TEXT_CHANNEL_RESOLVE');
```
you can't follow a news channel to another news channel i don't think

Reply:
correct, results in `DiscordAPIError: Cannot execute action on this channel type`